cmake/sca/codechecker/sca.cmake

```diff
@@ -0,0 +1,51 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (c) 2023, Basalte bv
+
+find_program(CODECHECKER_EXE CodeChecker REQUIRED)
+message(STATUS "Found CodeChecker: ${CODECHECKER_EXE}")
+
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+if(CODECHECKER_CONFIG)
+  message(STATUS "Found CodeChecker config: ${CODECHECKER_CONFIG}")
```

Comment:
(Optionally) would probably be good to check that the file exists here

Reply:
I'm changing this to a more generic way to pass options to the `analyze/parse` commands so the user is free to pass the config file.

Also the `cmake` currently doesn't take generated files into account. Change inc.